warehouse-management-service/internal/handler/handler.go

```diff
@@ -469,7 +479,226 @@ func (h *handler) DeleteShelfBlock(w http.ResponseWriter, r *http.Request) {
+		}
+	default:
+		{
+			h.logger.Log(log.Error, err)
+			h.response(w, http.StatusInternalServerError, api.ShelfResponse{Error: "Failed to get shelf"})
+		}
+	}
+}
+
+func (h *handler) CreateShelf(w http.ResponseWriter, r *http.Request) {
+	var createShelfRequest api.CreateShelfRequest
+
+	if r.Body == nil {
+		err := fmt.Errorf("request body cannot be empty")
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, api.ShelfResponse{
+			Error: err.Error()})
+		return
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	decoder.DisallowUnknownFields()
+	err := decoder.Decode(&createShelfRequest)
+	if err != nil {
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, api.ShelfResponse{
+			Error: "Failed to parse request"})
+		return
+	}
+
+	err = validator.Validate(createShelfRequest)
+	if err != nil {
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, api.ShelfResponse{
+			Error: fmt.Sprintf("Invalid input: %v", err.Error())})
+		return
+	}
+
+	shelf := wms.NewShelf(
+		createShelfRequest.Label,
+		createShelfRequest.Section,
+		createShelfRequest.Level,
+		createShelfRequest.ShelfBlockId)
+
+	err = h.shelfService.CreateShelf(r.Context(), shelf)
+	switch err {
+	case nil:
+		{
+			h.response(w, http.StatusOK, api.ShelfResponse{Message: fmt.Sprintf(
+				"Successfully created shelf: %s",
+				shelf.Id,
+			)})
+		}
+	case wms.InvalidShelfBlock:
+		{
+			h.logger.Log(log.Error, err)
+			h.response(w, http.StatusBadRequest, api.ShelfResponse{Error: fmt.Sprintf("%s: %s",
+				err.Error(),
+				shelf.ShelfBlockId,
+			)})
+		}
+	default:
+		{
+			h.logger.Log(log.Error, err)
+			h.response(
+				w,
+				http.StatusInternalServerError,
+				api.ShelfResponse{Error: "Failed to create shelf"},
+			)
+		}
+	}
+}
+
+func (h *handler) UpdateShelf(w http.ResponseWriter, r *http.Request) {
+	var updateShelfRequest api.UpdateShelfRequest
+
+	if r.Body == nil {
+		err := fmt.Errorf("request body cannot be empty")
+		h.logger.Log(log.Error, err)
+		h.response(
+			w,
+			http.StatusBadRequest,
+			api.ShelfResponse{Error: err.Error()})
+		return
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	decoder.DisallowUnknownFields()
+	err := decoder.Decode(&updateShelfRequest)
+	if err != nil {
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, api.ShelfResponse{
+			Error: "Failed to parse request",
+		})
+		return
+	}
+
+	err = validator.Validate(updateShelfRequest)
+	if err != nil {
+		h.logger.Log(log.Error, err)
+		h.response(w, http.StatusBadRequest, api.ShelfResponse{
+			Error: fmt.Sprintf("Invalid input: %v", err.Error())})
+		return
+	}
+
+	shelf := wms.Shelf{
+		Id:           updateShelfRequest.Id,
+		Label:        updateShelfRequest.Label,
+		Section:      updateShelfRequest.Section,
+		Level:        updateShelfRequest.Level,
+		ShelfBlockId: updateShelfRequest.ShelfBlockId,
+	}
+
+	err = h.shelfService.UpdateShelf(r.Context(), shelf)
+	switch err {
```

Comment:
Same comment as above. `if err != nil` is preferable

Reply:
Changed this to use if err != nil syntax as well